src/FixerConfiguration/FixerConfigurationResolver.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\FixerConfiguration;
+
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class FixerConfigurationResolver implements FixerConfigurationResolverInterface
+{
+    /**
+     * @var array<string, FixerOptionInterface>
+     */
+    private $options = array();
+
+    /**
+     * @var string|null
+     */
+    private $root;
+
+    /**
+     * @param FixerOptionInterface $option
+     *
+     * @throws \LogicException when the option is already defined
+     *
+     * @return $this
+     */
+    public function addOption(FixerOptionInterface $option)
+    {
+        if (array_key_exists($name = $option->getName(), $this->options)) {
+            throw new \LogicException(sprintf('The "%s" option is already defined.', $name));
+        }
+
+        $this->options[$name] = $option;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getOptions()
+    {
+        return array_values($this->options);
+    }
+
+    /**
+     * @param string $optionName
+     *
+     * @throws \LogicException when the option is already defined
+     *
+     * @return $this
+     *
+     * @deprecated will be removed in 3.0
+     */
+    public function mapRootConfigurationTo($optionName)
```

Comment:
Actually this would require `FixerConfigurationResolver` to not be `final`, is that ok?